mantle/kola/tests/coretest/core.go

```diff
@@ -378,3 +383,86 @@ func TestMachineID() error {
 	}
 	return nil
 }
+
+// TestRHCOSGrowpart tests whether rhcos-growpart was run successfully by checking
+// whether rhcos-growpart.service exited normally and root partition is >= 8 GB.
+func TestRHCOSGrowpart() error {
+	// check that rhcos-growpart.service was run and exited normally
+	execMainStatus, err := systemctlShow("ExecMainStatus", "rhcos-growpart.service")
+	if err != nil {
+		return err
+	}
+	if execMainStatus != "0" {
+		return fmt.Errorf("rhcos-growpart.service did not have exit code 0\n ExecMainStatus: %s", execMainStatus)
+	}
+	execMainPID, err := systemctlShow("ExecMainPID", "rhcos-growpart.service")
+	if err != nil {
+		return err
+	}
+	if execMainPID == "0" {
+		return fmt.Errorf("rhcos-growpart.service was not found")
+	}
+
+	// check that root partition is >= 8 GB
+	c := exec.Command("df", "/sysroot")
+	out, err := c.Output()
+	if err != nil {
+		return fmt.Errorf("Err: %s\n Out: %v", err, out)
+	}
+	dfOutput := strings.Split(string(out), "\n")
+	// verify header is as expected
+	if len(dfOutput) < 2 {
+		return fmt.Errorf("Expected at least two lines of df output: %q", out)
+	}
+	headerRegex := `Filesystem\s+1K-blocks\s+Used\s+Available\s+Use%\s+Mounted on`
+	if !regexp.MustCompile(headerRegex).MatchString(dfOutput[0]) {
+		return fmt.Errorf("df output has changed format: %q", dfOutput[0])
+	}
+	properties := strings.Fields(dfOutput[1])
+	if len(properties) != 6 {
+		return fmt.Errorf("Unexpected number of properties in df output: %q", dfOutput[0])
+	}
+	numBlocks, err := strconv.Atoi(properties[1])
+	if err != nil {
+		return fmt.Errorf("Error converting root partition size from string to int: %s", err)
+	}
+	// 8 GB is 8388608 1K-blocks
+	if numBlocks < 8388608 {
+		return fmt.Errorf("Root partition size is less than 8 GB, size in 1K-blocks: %d", numBlocks)
+	}
+
+	return nil
+}
+
+func systemctlShow(property string, service string) (string, error) {
+	c := exec.Command("systemctl", "show", "-p", property, "--value", service)
+	out, err := c.Output()
+	if err != nil {
+		return "", fmt.Errorf("Err: %s\n Out: %v", err, out)
+	}
+	result := strings.TrimRight(string(out), "\n")
+	return result, nil
+}
+
+// TestCoreOSGrowpart tests that coreos-growpart was run successfully by checking
+// whether ignition-ostree-growfs.service exited normally.
+func TestCoreOSGrowpart() error {
+	// MESSAGE_ID is SD_MESSAGE_UNIT_STOPPED from
+	// https://github.com/systemd/systemd/blob/master/src/systemd/sd-messages.h
+	c := exec.Command("journalctl", "-o", "json", "MESSAGE_ID=9d1aaa27d60140bd96365438aad20286",
+		"UNIT=ignition-ostree-growfs.service")
```

Comment:
Does this technique not work for `rhcos-growpart` as well?

Reply:
I just ran it again, strangely, I get no results if I use `exec.Command("journalctl", "-o", "json", "MESSAGE_ID=9d1aaa27d60140bd96365438aad20286", "UNIT=rhcos-growpart.service")`.
I'm trying with different `MESSAGE_ID`'s to see if that helps.

Reply:
Changed `MESSAGE_ID=` to  `SD_MESSAGE_UNIT_STARTED` (instead of `SD_MESSAGE_UNIT_STARTED`) and both FCOS and RHCOS now have results. As you mentioned, it should be due to `type=oneshot` units that `RemainAfterExit=yes` being considered "active". 
Additionally, I confirmed using an ignition drop-in that if the `growpart` service fails, querying the journal with `SD_MESSAGE_UNIT_STARTED` would yield no result.